cookbooks/bcpc-hadoop/recipes/hdfs_user_directories.rb

```diff
@@ -1,8 +1,15 @@
 Chef::Resource::RubyBlock.send(:include, Bcpc_Hadoop::Helper)
+Chef::Resource::RubyBlock.send(:include, Chef::Mixin::DeepMerge)
 
 ruby_block "hdfs_user_directories" do
   block do
-    # create the /user dirs: mode, dirs, home, perms
-    dir_creation('user', cluster_users + cluster_roles, '/user', '0700')
+
+    # Create directories for existing LDAP users and role accounts.
+    dirinfo = node.default['bcpc']['hadoop']['hdfs']['user']['dirinfo']
+    (cluster_users + cluster_roles).each do |user|
+      dirinfo[user] = deep_merge(dirinfo[user], { owner: user })
```

Comment:
As mentioned above, please clarify that `owner` will always be set to what is passed in the `name` of the resource.